src/gtclang/Frontend/GTClangPreprocessorAction.cpp

```diff
@@ -347,13 +411,46 @@ class GTClangLexer {
             consumeTokens(peekedTokens);
           }
         }
+        if(curToken.is(tok::identifier) &&
+           curToken.getIdentifierInfo()->getName() == "repository") {
+          if(stencilKind == SK_StencilFunction) {
+            reportError(token_.getLocation(),
+                        dawn::format("Illegal Expression: Declaration of repository as "
+                                     "stencil_function argument in '%s'.",
+                                     name));
+          } else {
+            std::string storedVariables = "";
+            // Accumulate all identifiers up to `;`
+            std::string repoStr;
+            if(peekAndAccumulateUntil(tok::semi, peekedTokens, repoStr)) {
+
+              // Split the comma separated string
+              llvm::SmallVector<StringRef, 5> allRepos;
+              StringRef(repoStr).split(allRepos, ',');
+              for(const auto& repo : allRepos) {
+                for(const auto& storageRepoPair : variableNametoRepoName_) {
+                  if(storageRepoPair.second == repo.str()) {
+                    storedVariables += "storage " + storageRepoPair.first + ";\n";
+                  }
+                }
+              }
+              if(!storedVariables.empty() && storedVariables[storedVariables.length() - 1 == '\n'])
```

Comment:
why `storedVariables[storedVariables.length() - 1 == '\n']`? I'm not quite sure what you try to accomplish here, do you mean `storedVariables[storedVariables.length() - 1] == '\n'` ;)?